File: 2026/04-April/20-April/solution.go
// 2078. Two Furthest Houses With Different Colors
func maxDistance(colors []int) int {
    n := len(colors)
    ans := 0

    for i := 0; i < n; i++ {
        if colors[i] != colors[0] {
            if i > ans {
                ans = i
            }
        }
        if colors[i] != colors[n-1] {
            if n-1-i > ans {
                ans = n-1-i
            }
        }
    }
 
    return ans
}

File: 2026/04-April/20-April/solution.kt
// 2078. Two Furthest Houses With Different Colors
class Solution {
    fun maxDistance(colors: IntArray): Int {
        val n = colors.size
        var ans = 0

        for (i in 0 until n) {
            if (colors[i] != colors[0]) {
                ans = maxOf(ans, i)
            } 
            if (colors[i] != colors[n - 1]) {
                ans = maxOf(ans, n - 1 - i)
            }
        }

        return ans
    }
}
